examples/network-hhmodel/network.rs

```diff
@@ -0,0 +1,102 @@
+use std::collections::HashSet;
+
+use crate::loader::{open_csv, HouseholdId, Id};
+use ixa::context::Context;
+use ixa::{define_edge_type, EdgeType};
+use ixa::{ContextNetworkExt, ContextPeopleExt, PersonId};
+use serde::Deserialize;
+
+define_edge_type!(HH, ());
+define_edge_type!(U5, ());
+define_edge_type!(U18, ());
+
+#[derive(Deserialize, Debug)]
+struct EdgeRecord {
+    v1: u16,
+    v2: u16,
+}
+
+fn create_household_networks(context: &mut Context, people: &[PersonId]) {
+    let mut households = HashSet::new();
+    for person_id in people {
+        let household_id = context.get_person_property(*person_id, HouseholdId);
+        if !households.contains(&household_id) {
+            households.insert(household_id);
+
+            let mut members = context.query_people((HouseholdId, household_id));
+            // create a dense network
+            while let Some(person) = members.pop() {
+                for other_person in &members {
+                    context
+                        .add_edge_bidi::<HH>(person, *other_person, 1.0, ())
+                        .unwrap();
+                }
+            }
+        }
+    }
+}
```

Comment:
You can use the return value of `insert` to avoid having to use `contains`

```suggestion
    for person_id in people {
        let household_id = context.get_person_property(*person_id, HouseholdId);
        if  households.insert(household_id) {
            let mut members = context.query_people((HouseholdId, household_id));
            // create a dense network
            while let Some(person) = members.pop() {
                for other_person in &members {
                    context
                        .add_edge_bidi::<HH>(person, *other_person, 1.0, ())
                        .unwrap();
                }
            }
        }
    }
}
```